### PR TITLE
OPS-753 AuthorizationGateway

### PR DESCRIPTION
**Pull request by tdonaworth**
## What changed

* Initial take on Authorization Gateway that can take a Authorization Provider
* minor lint/format cleanup
* added `pytest-mock` for use in testing
* added basic AuthorizationGateway unit test
* updated e2e to include better `a11y` logging output on violations.

## Issue

#753 

## How to test

* Currently only a framework, unit tests are the only existing tests.

### CODE REVIEW

File: frontend/cypress.config.js [hunk 1]
```diff
@@ -3,6 +3,22 @@ const { defineConfig } = require("cypress");
 module.exports = defineConfig({
     e2e: {
         baseUrl: "http://localhost:3000",
+        // Adding custom task logging, for better a11y output
+        // ref: https://docs.cypress.io/api/commands/task#Usage
+        // https://github.com/component-driven/cypress-axe#using-the-violationcallback-argument
+        setupNodeEvents(on, config) {
```
Comment (johndeange) [inline frontend/cypress.config.js:9]: Did you intend to check this in? (seems unrelated to the PR)

**Review by johndeange — APPROVED** [review]

File: frontend/cypress.config.js (hunk 1 above)
Comment (tdonaworth) [reply frontend/cypress.config.js:9]: Yep - it was more related to debugging the e2e failures, etc - in general it adds better logging output for a11y tests
